src/main/java/games/strategy/triplea/delegate/AbstractPlaceDelegate.java

```diff
@@ -204,23 +204,36 @@ public String placeUnits(final Collection<Unit> units, final Territory at, final
     if (error != null) {
       return error;
     }
-    // System.out.println("Placing " + MyFormatter.unitsToTextNoOwner(units) + " at " + at.getName() + " by " +
-    // player.getName());
     final List<Territory> producers = getAllProducers(at, m_player, units);
     producers.sort(getBestProducerComparator(at, units, m_player));
-    // System.out.println("Producers: " + producers);
     final IntegerMap<Territory> maxPlaceableMap = getMaxUnitsToBePlacedMap(units, at, m_player, true);
-    // System.out.println("Max Place Map: " + maxPlaceableMap);
-    final List<Unit> unitsLeftToPlace = new ArrayList<>(units);
-    unitsLeftToPlace.sort(getUnitConstructionComparator());
+
     // sort both producers and units so that the "to/at" territory comes first, and so that all constructions come first
     // this is because the PRODUCER for ALL CONSTRUCTIONS must be the SAME as the TERRITORY they are going into
-    for (final Territory producer : producers) {
-      if (unitsLeftToPlace.isEmpty()) {
-        break;
+    final List<Unit> unitsLeftToPlace = new ArrayList<>(units);
+    unitsLeftToPlace.sort(getUnitConstructionComparator());
+
+    while (!unitsLeftToPlace.isEmpty() && !producers.isEmpty()) {
+
+      // Get next producer territory
+      Territory producer = producers.get(0);
+      if (producers.size() > 1) {
+        producer = getRemotePlayer().selectProducerTerritoryForUnits(producers, at);
+        if (producer == null) {
+          break;
```

Comment:
When I cancel the "Select Producer Territory" dialog, the following message is displayed:

![pr-1766-cancel](https://cloud.githubusercontent.com/assets/4826349/26558416/83fb7f08-4476-11e7-8de8-3df48f2f8c19.png)

Not sure if this is the intended behavior.  It doesn't seem to have any harmful side effects, though.